components/builder-sessionsrv/src/server/handlers.rs

```diff
@@ -119,23 +129,11 @@ pub fn session_create(
     account_req.set_email(msg.take_email());
 
     match conn.route::<proto::AccountFindOrCreate, proto::Account>(&account_req) {
-        Ok(mut account) => {
-            let mut token = proto::SessionToken::new();
-            token.set_account_id(account.get_id());
-            token.set_extern_id(msg.get_extern_id());
-            token.set_provider(msg.get_provider());
-            token.set_token(msg.get_token().to_string().into_bytes());
-            let bytes = message::encode(&token)?;
-            let encoded_token = base64::encode(&bytes);
-
-            session.set_id(account.get_id());
-            session.set_email(account.take_email());
-            session.set_name(account.take_name());
-            session.set_token(encoded_token);
-            session.set_flags(flags.bits());
-            session.set_oauth_token(msg.take_token());
+        Ok(account) => {
+            let session = Session::build(msg, account, flags)?;
             {
-                state.sessions.write().unwrap().replace(session.clone());
+                debug!("issuing session, {:?}", session);
```

Comment:
should this debug go inside of Session::build ?

Reply:
I think it's best to debug in control flow and not units, that way you tell the story of what's happening in the order in which it's happening